frontend/src/ComponentsView.svelte

```diff
@@ -0,0 +1,336 @@
+<style>
+  button {
+    padding-top: 0.5em;
+    padding-bottom: 0.5em;
+    padding-left: 1em;
+    padding-right: 1em;
+  }
+
+  button:hover,
+  button:focus {
+    outline: none;
+    box-shadow: inset 1px 1px 0 var(--main-fg-color),
+      inset -1px -1px 0 var(--main-fg-color);
+  }
+
+  button:active {
+    box-shadow: inset 2px 2px 0 var(--main-fg-color),
+      inset -2px -2px 0 var(--main-fg-color);
+  }
+
+  .container {
+    min-height: 100%;
+    display: flex;
+    flex-direction: column;
+    flex-wrap: nowrap;
+    justify-content: center;
+    align-items: stretch;
+    align-content: center;
+  }
+
+  .inputs {
+    flex-grow: 1;
+    display: flex;
+    flex-direction: column;
+    flex-wrap: nowrap;
+    justify-content: flex-start;
+    align-items: stretch;
+  }
+
+  .inputs > *:first-child {
+    margin-top: 0;
+  }
+
+  .checkboxes {
+    display: flex;
+    flex-direction: column;
+    flex-wrap: nowrap;
+    justify-content: flex-start;
+    align-items: flex-start;
+    align-content: baseline;
+  }
+
+  .error {
+    margin-top: 2em;
+  }
+
+  .actions {
+    display: flex;
+    flex-direction: row;
+    flex-wrap: nowrap;
+    justify-content: space-evenly;
+    align-items: center;
+    align-content: center;
+    margin-top: 1em;
+  }
+
+  button,
+  select,
+  option {
+    background-color: var(--main-bg-color);
+    color: inherit;
+    border: 1px solid;
+    border-color: inherit;
+    border-radius: 0;
+    padding: 0.5em 1em;
+    margin-left: 0.5em;
+    margin-right: 0.5em;
+    font-size: inherit;
+    font-family: var(--font);
+    box-shadow: none;
+  }
+
+  select {
+    width: 100%;
+    margin: 0;
+    margin-bottom: 1em;
+  }
+
+  option {
+    font-family: monospace;
+  }
+
+  .filters {
+    display: flex;
+    flex-direction: row;
+    flex-wrap: nowrap;
+    justify-content: space-between;
+    align-items: flex-start;
+    align-content: center;
+  }
+
+  .filters > * {
+    flex-grow: 1;
+    margin: 0 1em;
+  }
+
+  .filters > *:first-child {
+    margin-left: 0;
+  }
+
+  .filters > *:last-child {
+    margin-right: 0;
+  }
+
+  hr {
+    border: 1px dashed var(--main-fg-color);
+    width: 100%;
+    margin: 2em 0;
+    padding: 0;
+    box-sizing: border-box;
+  }
+</style>
+
+<script>
+  import { selected, selectedResource } from "./stores.js";
+  import { splitAndCapitalize } from "./helpers.js";
+  import { onMount } from "svelte";
+
+  import ComponentConfigNode from "./ComponentConfigNode.svelte";
+  import LoadingText from "./LoadingText.svelte";
+  import Checkbox from "./Checkbox.svelte";
+
+  export let modifierView, resourceNodeDataMap;
+  let errorMessage,
+    all_components = true,
+    incl_analyzers = false,
+    incl_modifiers = false,
+    incl_packers = false,
+    incl_unpackers = false,
+    target_filter = null,
```

Comment:
JavaScript style is camelCase instead of Python's snake_case. It's kind of annoying to switch, and I'm not going to be a stickler about letting that block the PR. Just a heads up for the future.